Inc/ST-LIB_LOW/Communication/Server/Server.hpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Server.hpp
+ *
+ * Created on: Oct 12, 2023
+ * 		Author: Ricardo
+ */
+
+#pragma once
+#include "Communication/Ethernet/TCP/ServerSocket.hpp"
+#include "ErrorHandler/ErrorHandler.hpp"
+
+class Server{
+public:
+
+	enum ServerState{
+		RUNNING,
+		CLOSING,
+		CLOSED
+	};
+
+	ServerSocket *open_connection;
+	vector<ServerSocket*> running_connections;
+	IPV4 local_ip;
+	uint32_t local_port;
+	ServerState status;
+
+	static vector<Server*> running_servers;
```

Comment:
I'm not sure of using a `static vector`, because when  you add an element it might need to realocate memory, a more "_embedded_" approach would be to use a `static array` with some predefined size